concordium-std-derive/src/lib.rs

```diff
@@ -376,7 +382,27 @@ fn receive_worker(attr: TokenStream, item: TokenStream) -> syn::Result<TokenStre
 
     let fn_name = &ast.sig.ident;
     let rust_export_fn_name = format_ident!("export_{}", fn_name);
-    let wasm_export_fn_name = format!("{}.{}", contract_name, name);
+    let wasm_export_fn_name = format!("{}.{}", contract_name.value(), name.value());
+
+    // Validate the contract name independently to ensure that it doesn't contain a
+    // '.' as this causes a subtle error when receive names are being split.
+    let contract_name_validation =
+        ContractName::is_valid_contract_name(&format!("init_{}", contract_name.value()))
+            .map_err(|e| syn::Error::new(contract_name.span(), e.to_string()));
```

Comment:
This is minor, but you can just remove `.to_string()` since `::new` takes any value that implements `Display`, and thus calls `to_string` automatically.